src/e2e/java/teammates/e2e/cases/lnp/StudentEnrollmentLNPTest.java

```diff
@@ -0,0 +1,238 @@
+
+                dataBundle.instructors.forEach((key, instructor) -> {
+                    List<String> csvRow = new ArrayList<>();
+
+                    csvRow.add(instructor.googleId);
+                    csvRow.add("no");
+                    csvRow.add(instructor.courseId);
+
+                    // Create and add student enrollment data with a team number corresponding to each section number
+                    int dataLength = 75 * NUM_STUDENTS_PER_INSTRUCTOR;
+                    StringBuilder enrollData = new StringBuilder(dataLength);
+                    enrollData.append("\"Section|Team|Name|Email|Comments\n");
+
+                    for (int i = 0; i < NUM_STUDENTS_PER_INSTRUCTOR; i++) {
+                        String name = instructor.name + ".Student" + i;
+                        String email = instructor.name + ".Student" + i + "@gmail.tmt";
+
+                        enrollData.append(i / NUM_STUDENTS_PER_SECTION)
+                                .append('|')
+                                .append(i / NUM_STUDENTS_PER_SECTION)
+                                .append('|')
+                                .append(name)
+                                .append('|')
+                                .append(email)
+                                .append("|no comment\n");
+                    }
+                    enrollData.append('\"');
+                    csvRow.add(enrollData.toString());
+
+                    csvData.add(csvRow);
+                });
+
+                return csvData;
+            }
+        };
+    }
+
+    @Override
+    protected String getCsvConfigPath() {
+        return CSV_CONFIG_PATH;
+    }
+
+    @Override
+    protected String getJsonDataPath() {
+        return JSON_DATA_PATH;
+    }
+
+    @Override
+    protected int getNumberOfThreads() {
+        return NUM_INSTRUCTORS;
+    }
+
+    @Override
+    protected int getRampUpPeriod() {
+        return NUM_INSTRUCTORS * 2;
+    }
+
+    @Override
+    protected String getTestEndpoint() {
+        return Const.ResourceURIs.URI_PREFIX + Const.ResourceURIs.COURSE_ENROLL_SAVE + "?courseid=${courseId}";
+    }
+
+    @Override
+    protected String getTestMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    protected Map<String, String> getTestEndpointParameters() {
+        return new HashMap<>();
+    }
+
+    @Override
+    protected List<String> getTestEndpointPostVariables() {
+        List<String> args = new ArrayList<>();
+        args.add("${enrollData}");
+        return args;
+    }
+
+    @BeforeClass
+    public void classSetup() {
+        createTestData();
+        persistTestData(JSON_DATA_PATH);
+    }
+
+    @Test
+    public void runLnpTest() throws IOException {
+        runJmeter(false);
+    }
+
+    /**
+     * Removes the entities added for the student enrollment L&P test.
+     * There is no need to add the newly enrolled students to the JSON DataBundle#students. This is because the new
```

Comment:
This is implementation-specific, which is not what a header comment is for. Comments like this should be included in the method body itself.